strax/plugins/down_chunking_plugin.py

```diff
@@ -0,0 +1,40 @@
+import strax
+import types
+import inspect
+from .plugin import Plugin
+export, __all__ = strax.exporter()
+
+
+##
+# Plugin which allows to use yield in plugins compute method.
+# Allows to chunk down output before storing to disk.
+# Only works if multiprocessing is omitted.
+##
+
+@export
+class DownChunkingPlugin(Plugin):
+    """Plugin that merges data from its dependencies
+    """
+
+    def iter(self, iters, executor=None):
+
+        _plugin_uses_multi_threading = (self.parallel 
```

Comment:
Could we more simply assert/check that `not self.parallel` and `inspect.isgeneratorfunction(self.compute)` here? If compute is not a generator, you don't need this base class, and if it is, parallelization won't work and `parallel=False` seems right. 

If so, could do the checks in `__init__` (after `super.__init__` probably) since they don't depend on the executor anymore?